packages/fuels-core/src/types/native.rs

```diff
@@ -0,0 +1,447 @@
+use crate::{Bits256, Parameterize, Token, Tokenizable};
+use fuels_types::errors::Error;
+use fuels_types::param_types::{EnumVariants, ParamType};
+
+impl<const SIZE: usize, T: Parameterize> Parameterize for [T; SIZE] {
+    fn param_type() -> ParamType {
+        ParamType::Array(Box::new(T::param_type()), SIZE)
+    }
+}
+
+impl<const SIZE: usize, T: Tokenizable> Tokenizable for [T; SIZE] {
+    fn from_token(token: Token) -> Result<Self, Error>
+    where
+        Self: Sized,
+    {
+        let gen_error = |reason| {
+            Error::InvalidData(format!(
+                "While constructing an array of size {SIZE}: {reason}"
+            ))
+        };
+
+        match token {
+            Token::Array(elements) => {
+                let len = elements.len();
+                if len != SIZE {
+                    return Err(gen_error(format!(
+                        "Was given a Token::Array with wrong number of elements: {len}"
+                    )));
+                }
+
+                let detokenized = elements
+                    .into_iter()
+                    .map(Tokenizable::from_token)
+                    .collect::<Result<Vec<T>, _>>()
+                    .map_err(|err| {
+                        gen_error(format!(", not all elements could be detokenized: {err}"))
+                    })?;
+
+                Ok(detokenized.try_into().unwrap_or_else(|_| {
+                    panic!("This should never fail since we're checking the length beforehand.")
+                }))
+            }
+            _ => Err(gen_error(format!("Expected a Token::Array, got {token}"))),
+        }
+    }
+
+    fn into_token(self) -> Token {
+        Token::Array(self.map(Tokenizable::into_token).to_vec())
+    }
+}
```

Comment:
I love consistency :D With the rest of the types you first put all `Parametrize` then `Tokenizable` but here you put one after the other :smile:, also:
```suggestion
}

```

Reply:
consistency :D What an interesting way to say OCD :awesome 

I've changed it :)